src/compiler/checker.ts

```diff
@@ -6623,6 +6638,7 @@ namespace ts {
                     case SyntaxKind.ImportSpecifier:
                     case SyntaxKind.NamedImports:
                     case SyntaxKind.NamespaceImport:
+                    case SyntaxKind.NamespaceExport:

```

Comment:
Same issue around namespace exports not being locally visible - I don't think any code should need to hit this path.